# Video Game Sales Analysis

<p>Video games have become a significant part of the entertainment industry, generating billions in revenue each year. As the industry grows, understanding the factors that contribute to the success of video games is crucial for developers, publishers, and marketers alike. This analysis dives into historical video game sales data to uncover trends and patterns that can inform future decisions.</p>

<p>In this notebook, we will analyze video game sales data from various regions around the world. The dataset includes information about game titles, platforms, genres, publishers, and sales figures across different regions. Our goal is to answer key questions like:</p>

<ol>
    <li>What are the overall sales trends over the years?</li>
    <li>Which year had the highest global sales?</li>
    <li>Which platforms have generated the most global sales over time?</li>
    <li>Which genres and publishers have been the most popular in different years?</li>
    <li>Who are the top publishers consistently producing best-selling games?</li>
    <li>How do sales trends vary between regions such as North America, Europe, and Japan?</li>
    <li>What are the top 5 best-selling games of all time?</li>
    <li>What are the 5 least-selling games of all time?</li>
</ol>

<p>The first step in our analysis is to load the dataset and take a preliminary look at the data to understand its structure and content.</p>

## 1) Preparing for Analysis

### Import necessary libraries

In [29]:
import pandas as pd
from sqlalchemy import create_engine

### Importing Data and Creating SQL Database

In [31]:
csv_file_path = 'vgsales.csv'
df = pd.read_csv(csv_file_path)

engine = create_engine('sqlite:///vidoe_games_sales.db')

df.to_sql('sales', con=engine, if_exists='replace', index=False)

print("Data loaded successfully into the SQL table.")

Data loaded successfully into the SQL table.


### Display the first five rows

In [15]:
query = "SELECT * FROM sales" 

df_all_data = pd.read_sql(query, con=engine)

df_all_data.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

### Cleaning the data

#### Convert 'Year' column data type from FLOAT to INTEGER

In [61]:
df_all_data.loc[:, 'Year'] = df_all_data['Year'].astype(int)
df_all_data.to_sql('sales', con=engine, if_exists='replace', index=False)

16327

## 2) Analyzing the data

### 1. What are the overall sales trends over the years?
<p>The query analyzes video game sales across various years, excluding 2017 and 2020. By summing global sales for each year, it highlights the trends in game sales over time. This approach focuses on years with relevant data, offering a clear view of how game sales have evolved, while omitting the excluded years for a more accurate analysis.</p>

In [123]:
query = "SELECT Year, SUM(Global_Sales) AS 'Total Global Sales' FROM sales WHERE YEAR NOT IN (2017,2020) GROUP BY YEAR"

total_sales_by_year = pd.read_sql(query, con=engine)

total_sales_by_year

Year  Total Global Sales
0   1980               11.38
1   1981               35.77
2   1982               28.86
3   1983               16.79
4   1984               50.36
5   1985               53.94
6   1986               37.07
7   1987               21.74
8   1988               47.22
9   1989               73.45
10  1990               49.39
11  1991               32.23
12  1992               76.16
13  1993               45.98
14  1994               79.17
15  1995               88.11
16  1996              199.15
17  1997              200.98
18  1998              256.47
19  1999              251.27
20  2000              201.56
21  2001              331.47
22  2002              395.52
23  2003              357.85
24  2004              419.31
25  2005              459.94
26  2006              521.04
27  2007              611.13
28  2008              678.90
29  2009              667.30
30  2010              600.45
31  2011              515.99
32  2012              363.54
33  2013              368.11
34  2014              337.05
35  2015              264.44
36  2016               70.93

### 2. Which year had the highest global sales?
<p>In 2008, global sales reached their peak at 678.9 million dollars, surpassing every other year.</p>

In [134]:
query = "SELECT Year, MAX(Total_Global_Sales) AS 'Max Global Sales' FROM (SELECT Year, SUM(Global_Sales) AS Total_Global_Sales FROM sales WHERE YEAR NOT IN (2017,2020) GROUP BY YEAR)"

max_sales_year = pd.read_sql(query, con=engine)

max_sales_year

Year  Max Global Sales
0  2008             678.9

### 3. Which platforms have generated the most global sales over time?
<p>
The analysis of global sales data shows that the PlayStation 2 (PS2) stands out as the most successful gaming platform, with a staggering 1,233.46 million dollars in total global sales, far surpassing other consoles. Following closely, the Xbox 360 (X360) generated 969.61 million dollars, while the PlayStation 3 (PS3) achieved 949.35 million dollars in sales. The Wii also performed remarkably well, accumulating 909.81 million dollars in global sales. Rounding out the top five, the Nintendo DS (DS) secured 818.96 million dollars, underscoring the strong market presence of these platforms over time.
</p>

In [147]:
query = "SELECT Platform, SUM(Global_Sales) AS Total_Global_Sales FROM sales GROUP BY Platform ORDER BY Total_Global_Sales DESC LIMIT 10" 

total_sales_by_platform = pd.read_sql(query, con=engine)

total_sales_by_platform

Platform  Total_Global_Sales
0      PS2             1233.46
1     X360              969.61
2      PS3              949.35
3      Wii              909.81
4       DS              818.96
5       PS              727.39
6      GBA              313.56
7      PSP              291.71
8      PS4              278.10
9       PC              255.05

### 4. Which genres and publishers have been the most popular in different years?
<p>In the analysis of gaming trends, the Action genre emerged as the most dominant, with 2009 being the peak year, recording 272 releases. Following closely were the years 2012 and 2015, with 266 and 255 Action game releases, respectively. This trend underscores the genre's popularity during these periods.</p>

<p>On the publisher side, 2009 was a standout year for Activision, which led the industry with 121 game releases. Electronic Arts (EA) was also a significant player, particularly in the years 2008 and 2005, with 120 and 117 releases, respectively, solidifying its influence in the gaming market during those years.</p>

#### 1) Genre

In [181]:
query = "SELECT Year, Genre, MAX(Count_Genre) AS Max_Sales_Count FROM (SELECT Year, Genre, COUNT(Genre) AS Count_Genre FROM sales GROUP BY Year, Genre) GROUP BY YEAR ORDER BY Max_Sales_Count DESC LIMIT 10"

genre_by_year = pd.read_sql(query, con=engine)

genre_by_year

Year   Genre  Max_Sales_Count
0  2009  Action              272
1  2012  Action              266
2  2015  Action              255
3  2011  Action              239
4  2010  Action              226
5  2008  Action              221
6  2007  Action              211
7  2005  Action              192
8  2002  Sports              188
9  2014  Action              186

#### 2) Publisher

In [183]:
query_2 = "SELECT Year, Publisher, MAX(Count_Publisher) AS Max_Sales_Count FROM (SELECT Year, Publisher, COUNT(Publisher) AS Count_Publisher FROM sales GROUP BY Year, Publisher) GROUP BY YEAR ORDER BY Max_Sales_Count DESC LIMIT 10"

Publisher_by_year = pd.read_sql(query_2, con=engine)

Publisher_by_year

Year           Publisher  Max_Sales_Count
0  2009          Activision              121
1  2008     Electronic Arts              120
2  2005     Electronic Arts              117
3  2007     Electronic Arts              107
4  2006     Electronic Arts              102
5  2010          Activision               89
6  2002     Electronic Arts               88
7  2011             Ubisoft               87
8  2004     Electronic Arts               86
9  2012  Namco Bandai Games               85

### 5) Who are the top publishers consistently producing best-selling games?
<p>The analysis reveals that Nintendo is the leading publisher in producing best-selling games, with 427 titles generating an impressive total of 1,718.72 million dollars in global sales. Electronic Arts follows closely, with 597 best-selling games contributing 916.00 million dollars. Activision ranks third, with 292 titles generating 579.65 million dollars. Sony Computer Entertainment and Ubisoft round out the top five, with 260 and 229 best-selling games, respectively, and total sales of 519.93 million dollars and 352.26 million dollars. These publishers have consistently dominated the gaming market, driving substantial sales across their top-performing titles.</p>

In [203]:
query = "SELECT Publisher, COUNT(*) AS Best_Selling_Count, SUM(Global_Sales) AS Total_Sales FROM sales WHERE Global_Sales > (SELECT AVG(Global_Sales) FROM sales) GROUP BY Publisher ORDER BY Total_Sales DESC LIMIT 10"

best_selling_by_publishers = pd.read_sql(query, con=engine)

best_selling_by_publishers

Publisher  Best_Selling_Count  Total_Sales
0                      Nintendo                 427      1718.72
1               Electronic Arts                 597       916.00
2                    Activision                 292       579.65
3   Sony Computer Entertainment                 260       519.93
4                       Ubisoft                 229       352.26
5          Take-Two Interactive                 147       347.89
6        Microsoft Game Studios                  80       226.44
7                           THQ                 181       222.30
8                          Sega                 129       181.54
9  Konami Digital Entertainment                 116       166.93

### 6) How do sales trends vary between regions such as North America, Europe, and Japan?

<p>The analysis of sales trends across different regions reveals significant variations over time. In the early years (1980-1985), North America consistently led in sales, with a notable peak in 1981 at 33.40 million dollars, while Japan's market began to gain momentum, particularly from 1983 onwards, reaching a significant 14.56 million dollars by 1985. Throughout the 1990s, North America continued to dominate, especially in 1998 and 1999, with sales exceeding 120 million dollars each year. Meanwhile, Japan and Europe saw steady growth, with Japan peaking at 57.44 million dollars in 1996. The 2000s marked a period of strong sales across all regions, with North America peaking at 351.44 million dollars in 2008, Europe at 191.59 million dollars in 2009, and Japan remaining steady but significantly lower in comparison. However, by 2016, sales in all regions sharply declined, with North America and Europe dropping to around 22.66 and 26.76 million dollars, respectively, and Japan falling to 13.70 million dollars. The overall data indicates North America's dominance in the global market, followed by Europe, with Japan consistently contributing but at a lower scale.</p>

In [266]:
query = "SELECT Year, SUM(NA_Sales) AS NA_Total_Sales, SUM(EU_Sales) AS EU_Total_Sales, SUM(JP_Sales) AS JP_Total_Sales FROM sales GROUP BY Year"

regions_by_year = pd.read_sql(query, con=engine)

regions_by_year

Year  NA_Total_Sales  EU_Total_Sales  JP_Total_Sales
0   1980           10.59            0.67            0.00
1   1981           33.40            1.96            0.00
2   1982           26.92            1.65            0.00
3   1983            7.76            0.80            8.10
4   1984           33.28            2.10           14.27
5   1985           33.73            4.74           14.56
6   1986           12.50            2.84           19.81
7   1987            8.46            1.41           11.63
8   1988           23.87            6.59           15.76
9   1989           45.15            8.44           18.36
10  1990           25.46            7.63           14.88
11  1991           12.76            3.95           14.78
12  1992           33.87           11.71           28.91
13  1993           15.12            4.65           25.33
14  1994           28.15           14.88           33.99
15  1995           24.82           14.90           45.75
16  1996           86.76           47.26           57.44
17  1997           94.75           48.32           48.87
18  1998          128.36           66.90           50.04
19  1999          126.06           62.67           52.34
20  2000           94.49           52.75           42.77
21  2001          173.98           94.89           39.86
22  2002          216.19          109.74           41.76
23  2003          193.59          103.81           34.20
24  2004          222.59          107.32           41.65
25  2005          242.61          121.94           54.28
26  2006          263.12          129.24           73.73
27  2007          312.05          160.50           60.29
28  2008          351.44          184.40           60.26
29  2009          338.85          191.59           61.89
30  2010          304.24          176.73           59.49
31  2011          241.06          167.44           53.04
32  2012          154.96          118.78           51.74
33  2013          154.77          125.80           47.59
34  2014          131.97          125.65           39.46
35  2015          102.82           97.71           33.72
36  2016           22.66           26.76           13.70
37  2017            0.00            0.00            0.05
38  2020            0.27            0.00            0.00

### 7) What are the top 5 best-selling games of all time?
<p>The analysis of the top 5 best-selling games of all time is dominated by Nintendo titles. Leading the list is Wii Sports for the Wii, released in 2006, with an astounding 82.74 million dollars in global sales. Following this, the classic Super Mario Bros. for the NES, released in 1985, has generated 40.24 million dollars in sales. Mario Kart Wii, another Wii title, ranks third with 35.82 million dollars in sales since its 2008 release. Wii Sports Resort, also for the Wii, follows closely with 33.00 million dollars in sales since 2009. Finally, the iconic Pokemon Red/Pokemon Blue for the Game Boy, released in 1996, rounds out the list with 31.37 million dollars in sales. These figures underscore Nintendo's enduring influence and success in the gaming industry.</p>

In [214]:
query = "SELECT Name, Platform, Year, Genre, Publisher, Global_Sales FROM sales ORDER BY Global_Sales DESC LIMIT 5"

top_5_games = pd.read_sql(query, con=engine)

top_5_games

Name Platform  Year         Genre Publisher  \
0                Wii Sports      Wii  2006        Sports  Nintendo   
1         Super Mario Bros.      NES  1985      Platform  Nintendo   
2            Mario Kart Wii      Wii  2008        Racing  Nintendo   
3         Wii Sports Resort      Wii  2009        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB  1996  Role-Playing  Nintendo   

   Global_Sales  
0         82.74  
1         40.24  
2         35.82  
3         33.00  
4         31.37

### 8) What are the 5 least-selling games of all time?
<p>The analysis of the least-selling games shows that nearly 700 titles have only managed to reach 0.01 million dollars in global sales. These games span various platforms, genres, and publishers, but none have gained significant traction in the market. Among these games are Turok for the PC (2008, Touchstone), Coven and Labyrinth of Refrain for the PlayStation Vita (2016, Nippon Ichi Software), Super Battle For Money Sentouchuu: Kyuukyoku n... for the Nintendo 3DS (2016, Namco Bandai Games), Dragon Zakura DS for the Nintendo DS (2007, Electronic Arts), and Chameleon: To Dye For! for the Nintendo DS (2006, 505 Games). These are just a few of the many games that struggled to achieve substantial sales.</p>

In [247]:
query = "SELECT Name, Platform, Year, Genre, Publisher, Global_Sales FROM sales ORDER BY Global_Sales LIMIT 5"

least_5_games = pd.read_sql(query, con=engine)

least_5_games

Name Platform  Year   Genre  \
0                                              Turok       PC  2008  Action   
1                     Coven and Labyrinth of Refrain      PSV  2016  Action   
2  Super Battle For Money Sentouchuu: Kyuukyoku n...      3DS  2016  Action   
3                                   Dragon Zakura DS       DS  2007    Misc   
4                             Chameleon: To Dye For!       DS  2006  Puzzle   

              Publisher  Global_Sales  
0            Touchstone          0.01  
1  Nippon Ichi Software          0.01  
2    Namco Bandai Games          0.01  
3       Electronic Arts          0.01  
4             505 Games          0.01

## Conclusion 

<p>The analysis of video game sales data over the years provides valuable insights into the evolution of the gaming industry. Key trends reveal that the industry saw its highest global sales during the late 2000s, with North America leading the market. The PlayStation 2 emerged as the most successful platform, with genres like Action consistently topping the sales charts across various years. Nintendo stands out as a dominant force, not only with its popular platforms but also with top-selling games like Wii Sports and Super Mario Bros. Despite the success of major titles, a significant number of games struggled to achieve notable sales, highlighting the competitive nature of the industry.</p>